### DISCUSSION TITLE
Define an `eventBus` for a `bk-button`

### DISCUSSION THREAD

**Discussion by davidgarra** [mia-platform/community#356] [Q&A] [answered]
### Which section?

Backoffice

### Which is the version?

back-kit@1.4.7, microlc-backend@0.8.1, microlc-frontend@0.9.0

### Description

Is it possibile to define an `eventBus` for a `bk-button` inside the `customActions` of a `bk-table`?

I've my `bk-button` defined this way but I always get a `TypeError: e.pipe is not a function` error.

```json
  {
    "tag": "bk-button",
    "properties": {
      "eventBus": "eventBus.pool.ebtest",
      "content": "test",
      "config": {
        "events": {
          "label": "selected-data",
          "payload": { }
        }
      }
    }
  }
```

The `bk-button` is inside the property `customActions` of a `bk-table`

**Comment by clabene** [comment, marked as answer]
Hi @davidgarra ! Thanks for reaching out. Unfortunately, as of now, it is not possible to inject a custom event-bus to the components mounted within the `bk-table` via `customActions`. These components receive the same event-bus instance as the `bk-table`.

Nevertheless, feel free to share further details about your requirements for us to consider possible solutions.

As a side note, the configuration snippet you provided is probably missing the `action` property. You might want to update the configuration to:

```
  {
    "tag": "bk-button",
    "properties": {
      "content": "test",
      "action": {
        "type": "event",
        "config": {
          "events": {
            "label": "selected-data",
            "payload": {}
          }
        }
      }
    }
  }
```

**Reply by davidgarra** [reply]
ok, clear, thank you very much for the answer
